sdk/clientcore/core/src/main/java/io/clientcore/core/util/ClientLogger.java

```diff
@@ -308,6 +309,15 @@ private LoggingEventBuilder(Slf4jLoggerShim logger, LogLevel level, Map<String,
             this.globalPairs = globalContext;
         }
 
+        /**
+         * Returns true if this logging event will be logged.
+         *
+         * @return true if this logging event will be logged.
+         */
+        public boolean isEnabled() {
```

Comment:
Do we want this method to be public?

Reply:
yes, it's actually much more convenient to do

```java
var log = logger.atInfo(); // this costs nothing if `Info` is disabled
if (log.isEnabled()) {
  ..
}
log.log();
```

instead of

```java
if (logger.canLogAtLevel(INFORMATIONAL)) {
  log.atInfo()....
}
```

especially when you use some helper functions to populate properties and then need to carry both - the `log`  and the `canLog` flag.